design/stackhub.md

```diff
@@ -0,0 +1,207 @@
+# Collection transition to Stack Design
+
+## Key Concepts / Background
+- Simplify key concepts by eliminating Collections and Collection Hubs
+  -  Object model inheritence, e.g. A Collection ISA a Stack and a CollectionHub ISA a Stackhub, causes confusion especially for community messsaging, developers and new comers to the project.
+  -  Monolithic build (CollectionHub) causes painful lifecycle for making configuration or manage the curated stacks and pipelines.
+
+- In general, to replace Collection Hubs with Stack Hubs, and providing configuration for the governed stacks via the defined Kubernetes Custom Resources:  Kabanero CR.
+
+- Renaming the collection CR a stack CR, which has essentially the same data and format.
+
+- Although this is not a seemingly drastic change -- the concept of CollectionHub and Collections has tentacles throughout the website, guides, documentation, videos and blogs.
+
+## User stories
+
+- As the Kabanero community, we would like to simplify the messaging of Kabanero by removing the concept of CollectionHub and Collections.
+
+- As Champ (architect), I would like the definition, selection and build of Appsody stacks that are "blessed" untangled from the configuration of a specific Kabanero instance, moving from the Collection-Hub concept to a strictly Kubernetes-natural Kabanero and Stack CRD gitops user experience.
+
+## As-is
+
+- To introduce Kabanero topics, one must teach the student the concept of stacks, stack-hubs, pipelines, application development.  After all of that, then one must teach the student about collections and collection hubs, with a collection being a stack and a collection hub being a stack hub.  
+
+- To customize a stack, champ uses the appsody tools to copy the stack, and then has to manually convert it into a collection and include it in a collection hub.
+
+- To customize one character of a pipeline, I need to do a complete collection hub rebuild, which depending on how developer's are consuming collection hub releases may cause them to need to update all of their configuration -- for a change doesn't directly.
+
+## To-be
+-  Champ's (architect's) lifecycle changes to build stacks, using Appsody stackhub builds and independently building pipeline releases.
+-  Champ's experience is Kubernetes and operator-centric.  The configuration for Kabanero is defined solely by Kubernetes resources: `Kind:Kabanero` and `Kind:Stack`.
+-  Kabanero concepts are simplified to using stacks and their configuration.
+
+## Main Feature design
+
+### Custom Resource Changes
+
+#### Kind:Kabanero
+
+The Kabanero CRD will change to specifiy a Stack hub URL instead of a Collection Hub URL.  Previously, the Kabanero CRD did not concern itself with the relationship of pipelines with stacks.  The Kabanero CRD is enhanced to provide a default pipelines URL. 
```

Comment:
As per my earlier comment, I think the CRD should be referencing stack indexes rather than a stack hub.